daemon/endpoint.go

```diff
@@ -315,7 +315,7 @@ func (h *patchEndpointID) Handle(params PatchEndpointIDParams) middleware.Respon
 	}
 	ep.Mutex.Unlock()
 
-	if changed {
+	if ep.GetStateLocked() == endpoint.StateWaitingToRegenerate {
```

Comment:
Ah, I was just thinking that it should be in that state. We shouldn't be doing an `ep.GetStateLocked()` here because we don't currently hold the lock. I'll follow up.